program_budget/account_analytic_account.py

```diff
@@ -57,6 +57,7 @@ def create(self, cr, uid, vals, context=None):
 
         Create the related result if this account is under
         the root program node"""
+        context = context or {}
```

Comment:
I prefer to not loose previous empty context: `context = {} if context is None else context`